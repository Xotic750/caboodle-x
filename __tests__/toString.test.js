import {toString} from '../dist/caboodle-x';
import values from './helpers/nonSymbols';

/* istanbul ignore next */
const hasSymbol =
  typeof Symbol === 'function' && typeof Symbol('') === 'symbol';
/* istanbul ignore next */
const ifSymbolIt = hasSymbol ? it : xit;

describe('Basic tests', () => {
  it('should return a string for everything', () => {
    const expected = values.map(String);
    const actual = values.map(toString);
    expect(actual).toEqual(expected);
  });

  it('should throw for Object.create(null)', () => {
    expect(() => {
      toString(Object.create(null));
    }).toThrowErrorMatchingSnapshot();
  });

  ifSymbolIt('should throw for Symbol', () => {
    const sym = Symbol('foo');
    expect(() => {
      toString(sym);
    }).toThrowErrorMatchingSnapshot();

    const symObj = Object(sym);
    expect(() => {
      toString(Object(symObj));
    }).toThrowErrorMatchingSnapshot();
  });
});
