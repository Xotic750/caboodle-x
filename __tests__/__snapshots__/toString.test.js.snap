// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests should throw for Object.create(null) 1`] = `"Cannot convert object to primitive value"`;

exports[`Basic tests should throw for Symbol 1`] = `"Cannot convert a Symbol value to a string"`;

exports[`Basic tests should throw for Symbol 2`] = `"Cannot convert a Symbol value to a string"`;
